tests/server_decode.rs

```diff
@@ -0,0 +1,127 @@
+mod server_decode {
```

Comment:
This doesn't need a surrounding `mod`; it wraps tests out of the box already.

Reply:
as mentioned above, this for convenience of running just one set of tests — I use these like `describe` in other testing frameworks